### PR TITLE
docs(job-workers and workflows): Update job worker - broker concepts

### LINKED ISSUE #1972: Update docs sections 2.2 Workflows and 2.3 Job Workers to reflect current concepts
Docs sections 2.2 Workflows (https://docs.zeebe.io/basics/workflows.html) and 2.3 Job Workers (https://docs.zeebe.io/basics/job-workers.html) reference some older Zeebe concepts, e.g. "To start executing jobs, a worker must connect to the Zeebe broker declaring the specific job type it can handle. The Zeebe broker will then notify the worker whenever new jobs are created based on a publish-subsribe protocol. Upon receiving a notification, a worker performs the job and sends back a complete or fail message depending on whether the job could be completed successfully or not."

Need to edit these entires so they're up-to-date.

### PR DESCRIPTION
**Pull request by wints**
Some of the job worker - broker concepts described in the docs entires for 2.2 Workflows and 2.3 Job Workers are outdated. This PR updates the way these concepts are described and brings in new images where relevant. 

closes #1972

### CODE REVIEW

File: docs/src/bpmn-workflows/message-events.md [hunk 1]
```diff
@@ -11,7 +11,11 @@ A message can be referenced by one or more message events. It holds the informat
 * the name of the message
 * the correlation key
 
+<<<<<<< HEAD
```
Comment (npepinpe) [inline docs/src/bpmn-workflows/message-events.md]: I think this might be left overs from a merge conflict?

**Review by npepinpe — CHANGES_REQUESTED** [review]
I think there's some merge conflict leftovers in one of the files.

Otherwise looks good :+1:

File: docs/src/bpmn-workflows/message-events.md (hunk 1 above)
Comment (wints) [reply docs/src/bpmn-workflows/message-events.md]: Ach, you're right. Unstaged it. Thanks.

**Comment by wints** [comment]
@menski would you like to take a look at this or is it OK to go ahead and merge?

**Comment by menski** [comment]
@wints if @npepinpe is fine with your changes and approves the PR we can merge it, not sure if I have time to look at it soon

**Review by npepinpe — APPROVED** [review]
:+1:

**Review by menski — APPROVED** [review]

**Pull request merged**